crates/core/c8y_api/src/smartrest/operations.rs

```diff
@@ -28,8 +29,22 @@ pub struct Operations {
 impl Operations {
     pub fn add_operation(&mut self, operation: Operation) {
         self.operations.push(operation);
+        // as we insert, we need to maintain order, because depending on if `Operations` is created
+        // by adding operations one by one or by directory scan, we can end up with a different order
+        // TODO: refactor to provide more sane API, potentially using a backing `BTreeMap`
+        self.operations
+            .sort_unstable_by(|o1, o2| o1.name.cmp(&o2.name));
+    }
+
+    pub fn remove_operation(&mut self, name: &str) -> Option<Operation> {
+        self.operations.dedup();
+        let pos = self.operations.iter().position(|op| op.name == name);
+        pos.map(|pos| self.operations.remove(pos))
```

Comment:
If the switch to a `BTreeSet` is very minimal, I'm even okay with you making that switch in this PR itself, to avoid complications like this. Or if you prefer a standalone PR, that's also fine.

Reply:
Yes, better to switch to a `BTreeSet` right now.

Reply:
I would really prefer to avoid big changes to `Operations` struct in this PR because of parallel PR #3225, after one of these merges we expect to see conflicts which I'd like to minimise.
I'd also like to reorganize the module a bit in the refactor, so I'd rather do it as a separate PR.